### DISCUSSION TITLE
Issues installing executables

### DISCUSSION THREAD

**Discussion by adambor** [LNP-BP/nodes#2] [Support] [answered]
Running Ubuntu 22.04, rust 1.64.0-nightly

I've been trying to follow the demo from https://github.com/LNP-BP/nodes/blob/master/contrib/demo-rgb.sh . After running:
```
cargo +nightly -Z bindeps install lnpbp_nodes --version 0.8.0-rc.1
cargo +nightly -Z bindeps install lnpbp-cli --version 0.8.0-rc.1
```
I am not able to execute any of the downloaded executable:
```
ubuntu@ip-172-31-46-12:~$ lnpd
thread 'main' panicked at 'lnpd binary not found: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/lnpbp_nodes-0.8.0-rc.1/src/bin/lnpd.rs:8:10
stack backtrace:
   0: rust_begin_unwind
             at /rustc/23e21bdd25026e2839ebe946c2a937c1904887d2/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/23e21bdd25026e2839ebe946c2a937c1904887d2/library/core/src/panicking.rs:142:14
   2: core::result::unwrap_failed
             at /rustc/23e21bdd25026e2839ebe946c2a937c1904887d2/library/core/src/result.rs:1805:5
   3: lnpd::main
```

I tried to also use:
`cargo install descriptor-wallet —all-features`
Which successfully installed btc-cold and btc-hot - and I can run them both, but all the other executables give me above error...

**Comment by dr-orlovsky** [comment, marked as answer]
It seems I have overestimated abilities of new cargo `bindeps` feature https://rust-lang.github.io/rfcs/3028-cargo-binary-dependencies.html

It says:
> Note that cargo only supplies these dependencies when building your crate. If your program or library requires artifacts at runtime, you will still need to handle that yourself by some other means. Runtime requirements for installed crates are out of scope for this change.

It means that simplified way of installation with a single package like these `lnpbp_nodes`/`lnpbp-cli` can't exist and all components has to be installed separately :(

You can see the list of what have to be installed (with their versions) here

https://github.com/LNP-BP/nodes/blob/f4c16ef9cc05cae8c625a5a26e9bdcc81644a428/cli/Cargo.toml#L48-L56

and here

https://github.com/LNP-BP/nodes/blob/f4c16ef9cc05cae8c625a5a26e9bdcc81644a428/Cargo.toml#L60-L64

Just use `cargo install --all-features <package-name> --version <ver>` for each line in that list and that's all what can be done with existing cargo :(

**Reply by adambor** [reply]
Alright, installing it now with the separate commands, pasting here all the commands so it can be copy-pasted to docs or used by someone else:
```
cargo install descriptor-wallet --version "0.8.2"
cargo install bp-core --version "0.8.0"
cargo install rgb-std --version "0.8.0"
cargo install rgb20 --version "0.8.0-rc.3"
cargo install bp-cli --version "0.8.0-alpha.2"
cargo install lnp-cli --version "0.8.0"
cargo install rgb-cli --version "0.8.0-rc.1"
cargo install storm-cli --version "0.8.0-beta.1"
cargo install store-cli --version "0.8.0"
```

```
cargo install bp_node --version "0.8.0-alpha.2"
cargo install lnp_node --version "0.8.0"
cargo install rgb_node --version "0.8.0-rc.1"
cargo install storm_node --version "0.8.0-beta.1"
cargo install store_daemon --version "0.8.0"
```

EDIT: Created PR adding those commands to the demo file https://github.com/LNP-BP/nodes/pull/4
